src/wp-includes/option.php

```diff
@@ -2144,20 +2144,62 @@ function update_network_option( $network_id, $option, $value ) {
 	 */
 	$value = apply_filters( "pre_update_site_option_{$option}", $value, $old_value, $option, $network_id );
 
+	/*
+	 * To get the actual raw old value from the database, any existing pre filters need to be temporarily disabled.
+	 * Immediately after getting the raw value, they are reinstated.
+	 * The raw value is only used to determine whether a value is present in the database. It is not used anywhere
+	 * else, and is not passed to any of the hooks either.
+	 */
+	global $wp_filter;
+
+	$default_value = apply_filters( "default_site_option_{$option}", false, $option, $network_id );
+
+	$has_site_filter   = has_filter( "pre_site_option_{$option}" );
+	$has_option_filter = has_filter( "pre_option_{$option}" );
+	if ( $has_site_filter || $has_option_filter ) {
+		if ( $has_site_filter ) {
+			$old_ms_filters = $wp_filter[ "pre_site_option_{$option}" ];
+			unset( $wp_filter[ "pre_site_option_{$option}" ] );
+		}
+
+		if ( $has_option_filter ) {
+			$old_single_site_filters = $wp_filter[ "pre_option_{$option}" ];
+			unset( $wp_filter[ "pre_option_{$option}" ] );
+		}
+
+		$raw_old_value = is_multisite() ? get_network_option( $network_id, $option ) : get_option( $option, $default_value );
+
+		if ( $has_site_filter ) {
+			$wp_filter[ "pre_site_option_{$option}" ] = $old_ms_filters;
+		}
+		if ( $has_option_filter ) {
+			$wp_filter[ "pre_option_{$option}" ] = $old_single_site_filters;
+		}
+	} else {
+		$raw_old_value = $old_value;
+	}
+
```

Comment:
Alternatively, if we want to keep the pre filter and default filter logic separate, we could add a single `! is_multisite()` clause for the default filter handling here.